app/src/main/java/com/mdg/droiders/samagra/shush/AlarmScheduler.java

```diff
@@ -30,52 +36,299 @@ public AlarmScheduler(Context context) {
+     * @param endTime   The time of day at which phone is to be un-shushed
+     * @param days      A boolean array that is true if the alarm is to be
+     *                  scheduled for the corresponding day. i.e. days[0] is set to be true
+     *                  then a weekly alarm will be set such that it will shush the phone every Monday.
+     * @param rowID     Unique primary key of the shush alarm row.
+     */
+    public void setWeeklyAlarm(Calendar startTime, Calendar endTime, boolean[] days, Integer rowID) {
+        // To avoid various exceptions
+        if (rowID == null || days.length != 7) {
+            return;
+        }
+        // Instance of calendar that holds current system time
+        Calendar rightNow = Calendar.getInstance();
+
+        // Set YEAR, MONTH and DAY fields of startTime
+        // and endTime to today's YEAR, MONTH and DAY respectively
+        startTime.set(Calendar.YEAR, rightNow.get(Calendar.YEAR));
+        startTime.set(Calendar.MONTH, rightNow.get(Calendar.MONTH));
+        startTime.set(Calendar.DAY_OF_MONTH, rightNow.get(Calendar.DAY_OF_MONTH));
+        endTime.set(Calendar.YEAR, rightNow.get(Calendar.YEAR));
+        endTime.set(Calendar.MONTH, rightNow.get(Calendar.MONTH));
+        endTime.set(Calendar.DAY_OF_MONTH, rightNow.get(Calendar.DAY_OF_MONTH));
+        Log.d(LOG_TAG + "now", LOG_DATE_FORMAT.format(rightNow.getTime()));
+
+        // If startTime's hour is ahead of endTime's, assume that
+        // endTime is of that of next day
+        if (startTime.get(Calendar.HOUR_OF_DAY)
+                > endTime.get(Calendar.HOUR_OF_DAY)) {
+            endTime.add(Calendar.DAY_OF_MONTH, 1);
+        }
+
+        // If endTime is in the past increment both start and end time
+        if (endTime.compareTo(rightNow) < 0) {
+            startTime.add(Calendar.DAY_OF_MONTH, 1);
+            endTime.add(Calendar.DAY_OF_MONTH, 1);
+        }
+
+        // get Day index of start Time
+        int i = getDay(startTime);
+        for (int j = i; j < 7; j++) {
+            Log.d(LOG_TAG + "I-Val", String.valueOf(j));
+            if (days[j]) {
+                setAlarm(startTime.getTimeInMillis(),
+                        endTime.getTimeInMillis(),
+                        getStartDayID(j, rowID),
+                        getEndDayID(j, rowID));
+            }
+            // Update startTime and endTime
+            startTime.add(Calendar.DAY_OF_MONTH, 1);
+            endTime.add(Calendar.DAY_OF_MONTH, 1);
+        }
+        for (int j = 0; j < i; j++) {
+            startTime.add(Calendar.DAY_OF_MONTH, 1);
+            endTime.add(Calendar.DAY_OF_MONTH, 1);
+            Log.d(LOG_TAG + "I-Val", String.valueOf(j));
+            if (days[j]) {
+                setAlarm(startTime.getTimeInMillis(),
+                        endTime.getTimeInMillis(),
+                        getStartDayID(j, rowID),
+                        getEndDayID(j, rowID));
+            }
+        }
```

Comment:
Is there any reason for two loops with similar function, why can't they be merged into a single loop?

Reply:
Thanks for pointing this out.